### PR TITLE
minor refactor: Move Tab management into its own file

### PR DESCRIPTION
**Pull request by zadjii-msft**
I think we can all agree that `TerminalPage.cpp` is an unruly beast of a
file. It's got everything. It does everything. It can sometimes be a bit
hard to work with, because of simply how big it is. This PR tries to
alleviate this by making `TerminalPage.cpp` just a little smaller. It
does so by moving pretty much everything related to tab management into
its own file, `TabManagement.cpp`. These methods that have moved are all
the same as they were before, and they're still members of
`TerminalPage`. But now they're all in one place. 

I tried to move all the references to `_tabs` in `TerminalPage.cpp`, but
there's still a few that I left behind. Mostly because I felt that
moving those would be too gnarly a code change for an otherwise simple
cut&paste PR.

There are a few new methods I introduced:
* `_TabDragStarted` and `_TabDragCompleted`: These were lambdas before,
  promoted to full methods. 
* `_DismissTabContextMenus`: Remove all the right-click context menus
  from the tabs
* `_FocusCurrentTab`: This one's a bit trickier, we were actually doing
  this in a few different places, so I tried consolidating.
* `_HasMultipleTabs`: This doesn't need explaining.
* `_RemoveAllTabs`: Really, just encapsulation for the sake of removing
  a `_tabs` from `TerminalPage.cpp`
* `_ResizeTabContent`: Really, just encapsulation for the sake of
  removing a `_tabs` from `TerminalPage.cpp`

In the future, some enterprising young soul could try promoting that
file to its own class, and hiding `_tabs` (and `_mruTabs`) inside it.
Probably would need to take a reference to TerminalPage's `_tabView` and
`_newTabButton`. I'm not doing that right now, because I already hate
the idea of the ...

>  920 additions and 847 deletions.

... I'm making you look at already.

## Other thoughts

Some of the calls might be a little arbitrary - `_OpenNewTab` and
`_CreateNewTabFromSettings` probably should stay in `TerminalPage`? Or
at least elements of those might need to get split up better. Similarly
`TerminalPage::_OpenSettingsUI` stayed in `TerminalPage.cpp`, but it
does a lot of the same work as `_CreateNewTabFromSettings`. I'm not
saying this is the definitive places for these methods - it's code we're
working with, not stone ☺️

### CODE REVIEW

**Review by carlos-zamora — APPROVED** [review]
_Yeeeeaaaaah_. We should probably break up TerminalPage more haha.

Curious: does breaking it up help out with glomming & tab tear off at all? Maybe there's ways to shoe-horn some of that work in that effort?

If not, I say we file a CodeHealth task to figure out how to clean up TerminalPage a bit. Just to break up the responsibilities better.

**Comment by zadjii-msft** [comment]
> Curious: does breaking it up help out with glomming & tab tear off at all? Maybe there's ways to shoe-horn some of that work in that effort?

maybe? We're _way_ too far out to know for sure. Better encapsulation is always good though.

File: src/cascadia/TerminalApp/TabManagement.cpp [hunk 1]
```diff
@@ -0,0 +1,897 @@
...
+    bool TerminalPage::_HasMultipleTabs() const
+    {
+        return _tabs.Size() > 1;
+    }
+
+    void TerminalPage::_RemoveAllTabs()
+    {
+        // Since _RemoveTab is asynchronous, create a snapshot of the  tabs we want to remove
+        std::vector<winrt::TerminalApp::TabBase> tabsToRemove;
+        std::copy(begin(_tabs), end(_tabs), std::back_inserter(tabsToRemove));
```
Comment (DHowett) [inline src/cascadia/TerminalApp/TabManagement.cpp:883]: clever

**Review by DHowett — APPROVED** [review]

**Comment by DHowett** [comment]
Conflict resolution req'd.

**Pull request merged**